### DISCUSSION TITLE
Error in the workflow

### DISCUSSION THREAD

**Discussion by uoojin95** [lowlighter/metrics#130] [Q&A] [answered]
Hi I'm not really sure why an error occurs within the work flow. 
Seems to be the graphql query that is failing.

this is the error log

────────────────────────────────────────────────────────────────
An error occured, logging debug message :
metrics/compute/uoojin1 > start
{
  languages: true,
  isocalendar: true,
  stars: true,
  'languages.ignored': [ 'Jupyter Notebook' ],
  'languages.skipped': [],
  'isocalendar.duration': 'half-year',
  'stars.limit': 4,
  base: false,
  'base.header': true,
  'base.activity': true,
  'base.community': true,
  'base.repositories': true,
  'base.metadata': true,
  'config.timezone': 'America/Toronto',
  'config.output': 'svg',
  'config.animations': true,
  'config.padding': '6%',
  repositories: 100,
  template: 'classic'
}
metrics/compute/uoojin1 > graphql query
Error: Something went wrong while executing your query. Please include `0601:77AA:22CD5C0:282BEE0:5FF6B8FA` when reporting this issue.
    at /metrics/node_modules/@octokit/graphql/dist-node/index.js:69:13
    at processTicksAndRejections (node:internal/process/task_queues:93:5)
    at async metrics (file:///metrics/source/app/metrics.mjs:46:35)
    at async file:///metrics/source/app/action/index.mjs:233:28 {
  errors: [
    {
      message: 'Something went wrong while executing your query. Please include `0601:77AA:22CD5C0:282BEE0:5FF6B8FA` when reporting this issue.'
    }
  ],
  headers: {
    'access-control-allow-origin': '*',
    'access-control-expose-headers': 'ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, Deprecation, Sunset',
    'cache-control': 'no-cache',
    connection: 'close',
    'content-encoding': 'gzip',
    'content-security-policy': "default-src 'none'",
    'content-type': 'application/json; charset=utf-8',
    date: 'Thu, 07 Jan 2021 07:32:11 GMT',
    'referrer-policy': 'origin-when-cross-origin, strict-origin-when-cross-origin',
    server: 'GitHub.com',
    status: '200 OK',
    'strict-transport-security': 'max-age=31536000; includeSubdomains; preload',
    'transfer-encoding': 'chunked',
    vary: 'Accept-Encoding, Accept, X-Requested-With, Accept-Encoding',
    'x-accepted-oauth-scopes': 'repo',
    'x-content-type-options': 'nosniff',
    'x-frame-options': 'deny',
    'x-github-media-type': 'github.v3; format=json',
    'x-github-request-id': '0601:77AA:22CD5C0:282BEE0:5FF6B8FA',
    'x-oauth-scopes': '',
    'x-ratelimit-limit': '5000',
    'x-ratelimit-remaining': '4997',
    'x-ratelimit-reset': '1610007590',
    'x-ratelimit-used': '3',
    'x-xss-protection': '1; mode=block'
  },
  request: {
    query: 'query Metrics {\n' +
      '  user(login: "uoojin1") {\n' +
      '    databaseId\n' +
      '    name\n' +
      '    login\n' +
      '    createdAt\n' +
      '    avatarUrl\n' +
      '    websiteUrl\n' +
      '    isHireable\n' +
      '    twitterUsername\n' +
      '    repositories(last: 0, isFork: false, ownerAffiliations: OWNER) {\n' +
      '      totalCount\n' +
      '      totalDiskUsage\n' +
      '      nodes {\n' +
      '        name\n' +
      '      }\n' +
      '    }\n' +
      '    packages {\n' +
      '      totalCount\n' +
      '    }\n' +
      '    starredRepositories {\n' +
      '      totalCount\n' +
      '    }\n' +
      '    watching {\n' +
      '      totalCount\n' +
      '    }\n' +
      '    sponsorshipsAsSponsor {\n' +
      '      totalCount\n' +
      '    }\n' +
      '    sponsorshipsAsMaintainer {\n' +
      '      totalCount\n' +
      '    }\n' +
      '    contributionsCollection {\n' +
      '      totalRepositoriesWithContributedCommits\n' +
      '      totalCommitContributions\n' +
      '      restrictedContributionsCount\n' +
      '      totalIssueContributions\n' +
      '      totalPullRequestContributions\n' +
      '      totalPullRequestReviewContributions\n' +
      '    }\n' +
      '    calendar:contributionsCollection(from: "2020-12-24T07:32:10.474Z", to: "2021-01-07T07:32:10.474Z") {\n' +
      '      contributionCalendar {\n' +
      '        weeks {\n' +
      '          contributionDays {\n' +
      '            color\n' +
      '          }\n' +
      '        }\n' +
      '      }\n' +
      '    }\n' +
      '    repositoriesContributedTo {\n' +
      '      totalCount\n' +
      '    }\n' +
      '    followers {\n' +
      '      totalCount\n' +
      '    }\n' +
      '    following {\n' +
      '      totalCount\n' +
      '    }\n' +
      '    issueComments {\n' +
      '      totalCount\n' +
      '    }\n' +
      '    organizations {\n' +
      '      totalCount\n' +
      '    }\n' +
      '  }\n' +
      '}\n'
  }
}
Error: Process completed with exit code 1.

**Comment by lowlighter** [comment]
Hello !

Actually I'm wondering what's happening too...
There is indeniabely an error but I'm not able to replicate it.

I forked your repo at [lowlighter/uoojin1](https://github.com/lowlighter/uoojin1) and I seem able to generate your metrics when I provide a personal token (by the way, this [line](https://github.com/uoojin1/uoojin1/blob/main/.github/workflows/metrics.yml#L6) should have `main`  instead of `master` but that's a detail) and ran it multiple times, so there no issues with your workflows.

Using the [GraphQL explorer](https://docs.github.com/en/free-pro-team@latest/graphql/overview/explorer) with generated query seems ok too.

<details>
<summary>Generated query</summary>

```
query Metrics {
  user(login: "uoojin1") {
    databaseId
    name
    login
    createdAt
    avatarUrl
    websiteUrl
    isHireable
    twitterUsername
    repositories(last: 0, isFork: false, ownerAffiliations: OWNER) {
      totalCount
      totalDiskUsage
      nodes {
        name
      }
    }
    packages {
      totalCount
    }
    starredRepositories {
      totalCount
    }
    watching {
      totalCount
    }
    sponsorshipsAsSponsor {
      totalCount
    }
    sponsorshipsAsMaintainer {
      totalCount
    }
    contributionsCollection {
      totalRepositoriesWithContributedCommits
      totalCommitContributions
      restrictedContributionsCount
      totalIssueContributions
      totalPullRequestContributions
      totalPullRequestReviewContributions
    }
    calendar:contributionsCollection(from: "2020-12-24T08:33:27.521Z", to: "2021-01-07T08:33:27.521Z") {
      contributionCalendar {
        weeks {
          contributionDays {
            color
          }
        }
      }
    }
    repositoriesContributedTo {
      totalCount
    }
    followers {
      totalCount
    }
    following {
      totalCount
    }
    issueComments {
      totalCount
    }
    organizations {
      totalCount
    }
  }
}
```

</details>

Anyway the same query is executed on [shared instance](https://metrics.lecoq.io) so I really don't know what could cause this error.

By searching on google, it seems that `Something went wrong while executing your query`  pretty much means that error in on GitHub's end.

What you could try though (although I'm not sure it'll have any effect) : 
- Regenerate a new personal token
- If you use a scope-less token, maybe try to add wider permissions 
- You can also confirm that you're able to query above on [GraphQL explorer](https://docs.github.com/en/free-pro-team@latest/graphql/overview/explorer)

Seems you already find a workaround for now 🙂

If nothing above works or does not give additional data, I'll try to contact GitHub support with your requests ids to get more informations. 

Have a nice day !

**Comment by uoojin95** [comment]
Hey lowlighter!

Thanks for the detailed response. It was really helpful and I was able to get it to work.

I think it was a scope issue. I added few more permissions to the token and it started working :).

One weird thing I noticed is that when I generate the svg from this website (https://metrics.lecoq.io/), it picks up my contributions to repositories that are from a private organization (need SSO login). However when I run the query with the github action, it doesn't seem to pick up those contributions (so less commits are shown on the calendar plugin).

Please let me know if there is a way to fix this issue

Thank you!

**Comment by lowlighter** [comment]
Glad you were able to make it work !

Do you know which scopes were actually missing ? 
Maybe it can be useful if others encounter the same problem.

___

As for the contributions from private organizations, I'm not sure myself...

Playing the following query on [GraphQL explorer](https://docs.github.com/en/free-pro-team@latest/graphql/overview/explorer) and executing the JS one-liner :

![image](https://user-images.githubusercontent.com/22963968/104014526-75115c80-51b3-11eb-9bfd-4ae96a3eb2b3.png)

<details>
<summary>Query and JS</summary>

```graphql
query Calendar {
  user(login: "uoojin1") {
    calendar:contributionsCollection(from: "2020-01-07T00:00:00.000Z", to: "2021-01-08T00:00:00.000Z") {
      contributionCalendar {
        weeks {
          contributionDays {
            contributionCount
            color
            date
          }
        }
      }
    }
  }
}
```

```js
Object(/* Pasted result */).data.user.calendar.contributionCalendar.weeks.reduce((a, b) => a + b.contributionDays.reduce((c, d) => c + d.contributionCount, 0), 0)
```

</details>

Yields me more or less the same result which is displayed on your profile page : 
![image](https://user-images.githubusercontent.com/22963968/104015958-adb23580-51b5-11eb-9b99-c82afd974308.png)

Basically that's what [metrics.lecoq.io](https://metrics.lecoq.io) is able to display since it only uses a scope-less bot token. 

___

I confess that it's weird that your personal token seems to be able to do less than a stranger token x)
Maybe there's some strange behaviour with GitHub API depending on which scopes are enabled.

You can try the following workaround, but I think it's a bit troublesome so maybe it's simpler to use [metrics.lecoq.io](https://metrics.lecoq.io).

Basically, what you could do is to generate isocalendar in its own step (a bit like [here](https://github.com/lowlighter/lowlighter/blob/master/.github/workflows/metrics.yml#L372-L384)) with a scope-less token (so basically you would use a second token).

If you get the same results or if you encounter again the initial issue (`Something went wrong with query`), then you'll need to use a token from someone else instead (like a token from a bot account) and use the `user: uoojin1` option so it'll compute your metrics and not those of the bot account.

It's like reproducing the same behaviour from [metrics.lecoq.io](https://metrics.lecoq.io) so I'm pretty confident it would work, but seems a bit hacky.

Sorry that you got so much trouble so far. As I don't really work with GitHub organizations myself, it's something that I don't really thoughrouly tested 😅 

Hope you still appreciate it 🙂

**Comment by uoojin95** [comment, marked as answer]
yeah it was a little weird haha 😅.

I was actually able to get it fully working with a new token (scopeless) and authorizing it for use with SSO.
[Authorizing a personal access token for use with SAML single sign-on](https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/authorizing-a-personal-access-token-for-use-with-saml-single-sign-on)

Took a long time but finally got there lol.

Thanks for all the help, really appreciate it!

**Comment by lowlighter** [comment]
Nice ! Thanks for linking the documentation too !
